### PR TITLE
User sarath

### PR DESCRIPTION
**Pull request by sarath-rp**

### CODE REVIEW

File: ZohoCRMCharts/Classes/ZCRMWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+/**
+Data object for KPI components of all types.
+*/
+public class ZCRMKPIRow {
+	
+	internal var value: String!
+	public var difference: String!
+	public var rate: String!
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMWrapper.swift]: properties shouldn't be public.

File: Example/ZohoCRMCharts.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+				);
+				GCC_PREPROCESSOR_DEFINITIONS = (
+					"DEBUG=1",
+					"$(inherited)",
+				);
+				INFOPLIST_FILE = Tests/Info.plist;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = "org.cocoapods.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_SWIFT3_OBJC_INFERENCE = Default;
```
Comment (BoopathyPSiva) [inline Example/ZohoCRMCharts.xcodeproj/project.pbxproj:360]: we should support from 3.0

File: Example/ZohoCRMCharts.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+					"$(inherited)",
+				);
+				GCC_SYMBOLS_PRIVATE_EXTERN = NO;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.3;
```
Comment (BoopathyPSiva) [inline Example/ZohoCRMCharts.xcodeproj/project.pbxproj]: check configured iOS minimum deployment target.

File: ZohoCRMCharts/Classes/ZCRMCommonUtil.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	var type: ZCRMKPIComponent! { get set}
+	var isScorecard: Bool { get }
+	var isRankings: Bool { get }
+	var isBasic: Bool { get }
+	var isStandard: Bool { get }
+	var isGrowthIndex: Bool { get }
+}
+
+extension KPIUtil {
+
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMCommonUtil.swift:19]: Is this really needed?

File: ZohoCRMCharts/Classes/ZCRMCommonUtil.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+//  ZohoCRMCharts
+//
+//  Created by Sarath Kumar Rajendran on 24/10/18.
+//
+
+protocol KPIUtil {
+	
+	var type: ZCRMKPIComponent! { get set}
+	var isScorecard: Bool { get }
+	var isRankings: Bool { get }
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMCommonUtil.swift:12]: is this really needed?

File: ZohoCRMCharts/Classes/ZCRMKPI.swift [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+internal extension ZCRMKPI {
+	
+	/**
+		Renders the view, related to its type.
+	*/
+	private func render() {
+		
+		self.clipsToBounds = true
+		self.layoutIfNeeded()
+		self.backgroundColor = UIColor(red: 182/255.0, green: 186/255.0, blue: 193/255.0, alpha: 1)
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMKPI.swift]: set same value for RGB

File: ZohoCRMCharts/Classes/ZCRMKPI.swift [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+	
+	/**
+		Adds the constraints for the simple KPI.
+	*/
+	private func addSimpleKpiConstraints() {
+		
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[value]", options: [], metrics: nil, views: ["title": self.titleView, "value": valueView])
+		if !self.isBasic {
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-16-[value(==comparedTo)]-[comparedTo]-15-|", options: [.alignAllBottom], metrics: nil, views: ["value": valueView, "comparedTo": self.comparedToView])
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMKPI.swift]: it should be 15

File: ZohoCRMCharts/Classes/ZCRMKPI.swift [hunk 8]
```diff
@@ -0,0 +1,513 @@
...
+		}
+	}
+	
+	
+	/**
+		Sets the compared to text for standard, growth index and scorecard KPI.
+	*/
+	private func updateComparedToViewText(text: String?) {
+		
+		var textToFit = ""
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMKPI.swift]: Use String()

File: ZohoCRMCharts/Classes/ZCRMKPI.swift (hunk 7 above)
Comment (sarath-rp) [reply ZohoCRMCharts/Classes/ZCRMKPI.swift]: done

File: ZohoCRMCharts/Classes/ZCRMKPI.swift (hunk 8 above)
Comment (sarath-rp) [reply ZohoCRMCharts/Classes/ZCRMKPI.swift]: Both are same. why we shouldn't use that?

Comment (sarath-rp) [reply ZohoCRMCharts/Classes/ZCRMKPI.swift]: done

File: ZohoCRMCharts/Classes/ZCRMKPICell.swift [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+	
+	private func renderView() {
+		
+		//adding subviews
+		self.rowLable.translatesAutoresizingMaskIntoConstraints = false
+		self.valueLabel.translatesAutoresizingMaskIntoConstraints = false
+		self.rateLabel.translatesAutoresizingMaskIntoConstraints = false
+		self.rateBar.translatesAutoresizingMaskIntoConstraints = false
+		self.rightContainer.translatesAutoresizingMaskIntoConstraints = false
+		self.rowLable.adjustsFontSizeToFitWidth = true
```
Comment (BoopathyPSiva) [inline ZohoCRMCharts/Classes/ZCRMKPICell.swift]: Remove this. we don't need to change the KPIRenderOptions.

**Pull request closed without merging**